stan/math/prim/scal/prob/logistic_rng.hpp

```diff
@@ -3,26 +3,37 @@
 
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
 #include <stan/math/prim/scal/err/check_positive_finite.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
-#include <stan/math/prim/scal/fun/log1p.hpp>
-#include <stan/math/prim/scal/meta/length.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
 #include <stan/math/prim/scal/meta/VectorBuilder.hpp>
-#include <stan/math/prim/scal/meta/return_type.hpp>
-#include <stan/math/prim/scal/meta/include_summand.hpp>
 #include <boost/random/exponential_distribution.hpp>
 #include <boost/random/variate_generator.hpp>
 #include <string>
 
 namespace stan {
   namespace math {
-
-    template <class RNG>
-    inline double
-    logistic_rng(double mu,
-                 double sigma,
+    /**
+     * Return a pseudorandom Logistic variate for the given location and scale
+     * using the specified random number generator.
+     *
+     * mu and sigma can each be either scalars or vectors. All vector inputs
+     * must be the same length.
+     *
+     * @tparam T_loc Type of location parameter
+     * @tparam T_scale Type of scale parameter
+     * @tparam RNG type of random number generator
+     * @param mu (Sequence of) location parameter(s)
+     * @param sigma (Sequence of) scale parameter(s)
+     * @param rng random number generator
+     * @return Logistic random variate
+     * @throw std::domain_error if mu is infinite or sigma is nonpositive
+     * @throw std::invalid_argument if vector arguments are not the same length
+     */
+    template <typename T_loc, typename T_scale, class RNG>
+    inline typename VectorBuilder<true, double, T_loc, T_scale>::type
+    logistic_rng(const T_loc& mu,
+                 const T_scale& sigma,
```

Comment:
[optional]

pack arguments up.  I'm going to stop repeating this, but it'd be great for me if it got fixed everywhere.